### DISCUSSION TITLE
How do proxies work?

### DISCUSSION THREAD

**Discussion by SanicBTW** [ppy/osu-framework#6472] [Q&A] [answered]
I'm currently trying to work with proxies but I can't really get to work with them, I kinda expected them to be able to detach from its original form and then be able to go back to it, like some kind of `DestroyProxy` or something to destroy said proxy and return the rendering to the original drawable.

**Comment by peppy** [comment]
This isn't supported currently.

Maybe you could start by explaining your use case.

**Reply by SanicBTW** [reply]
So my use case was to make a copy of an existing sprite, change some values and when I was done with it, I would delegate the control back to the original sprite, like some kind of a leased bindable? 

To be more specific, I wanted to make a copy of a sprite that would change its size and then after some transforms I would dispose said copy and the original sprite would recover the rendering capabilities, behaving like if nothing that happened with the copy reflected it.

**Reply by peppy** [reply]
that's not what proxies are for. you're probably best just creating a second sprite, or setting up some custom bindable flow.

**Reply by SanicBTW** [reply]
yeahh I thought of it for a second and I discarded the idea, but I would like to know what are the use cases for proxies though

**Reply by smoogipoo** [reply, marked as answer]
The use case is to draw things at a different level to where they're physically located within the draw tree. It's a very specific scenario where you have a deep structure that is used to control sizing in a very specific way, but some nested component actually needs to be an overlay (like a hit explosion).

**Reply by SanicBTW** [reply]
Ohhh, I get it now, thanks!
